### PR TITLE
syncs stored presets to latest local

### PR DESCRIPTION
**Pull request by Rcm37**
This should solve our issues with older version of saved presets having outdated keys. We can change the upload logic later to only store label + breakdownNotes but for right now its not hurting anything to save extra data.

### CODE REVIEW

File: src/components/PresetSection/presetSectionApi.ts [hunk 1]
```diff
@@ -1,9 +1,48 @@
...
+    inventorySlots: [],
+    equipmentSlots: []
+  };
+  //create a map of all item labels to their default object
+  let itemDataMap = new Map();
+  await itemData.forEach(element => {
+    itemDataMap.set(element.label, element);
+  });
+  //loop stored preset inventory items
+  for(let i = 0 ; i < stored.inventorySlots.length ; i++) {
```
Comment (cdtinney) [inline src/components/PresetSection/presetSectionApi.ts:29]: btw if you install ESLint/Prettier it'll auto-fix formatting on save (e.g. adding space after `for`, `if`)

File: src/components/PresetSection/presetSectionApi.ts [hunk 2]
```diff
@@ -1,9 +1,48 @@
...
+    let itemLabel = stored.inventorySlots[i].label;
+    let defaultItem = {...itemDataMap.get(itemLabel)};
+    if(stored.inventorySlots[i].breakdownNotes != "") {
```
Comment (cdtinney) [inline src/components/PresetSection/presetSectionApi.ts:32]:
```suggestion
    if(stored.inventorySlots[i].breakdownNotes !== "") {
```

File: src/components/PresetSection/presetSectionApi.ts [hunk 3]
```diff
@@ -1,9 +1,48 @@
...
+      defaultItem.breakdownNotes = stored.inventorySlots[i].breakdownNotes
+    }
+    newPreset.inventorySlots[i] = defaultItem;
+  }
+  //loop stored equipment items
+  for(let i = 0 ; i < stored.equipmentSlots.length ; i++) {
+    let itemLabel = stored.equipmentSlots[i].label;
+    let defaultItem = {...itemDataMap.get(itemLabel)};
+    if(stored.equipmentSlots[i].breakdownNotes != "") {
```
Comment (cdtinney) [inline src/components/PresetSection/presetSectionApi.ts:41]:
```suggestion
    if(stored.equipmentSlots[i].breakdownNotes !== "") {
```

File: src/components/PresetSection/presetSectionApi.ts (hunk 1 above)
Comment (Rcm37) [reply src/components/PresetSection/presetSectionApi.ts:29]: weird i have both installed

**Pull request merged**